app/bundles/UserBundle/Controller/PublicController.php

```
@@ -139,4 +141,103 @@ public function passwordResetConfirmAction(Request $request, UserPasswordHasherI
+
+                return $this->delegateView([
+                    'viewParameters' => [
+                        'form' => $form->createView(),
+                    ],
+                    'contentTemplate' => '@MauticUser/Security/register.html.twig',
+                    'passthroughVars' => [
+                        'route' => $action,
+                    ],
+                ]);
+            }
+
+            if ($form->isSubmitted() && $form->isValid()) {
+                try {
+                    $formData = $form->getData();
+
+                    // Create user from form data
+                    $user = new User();
+                    $user->setUsername($formData['username']);
+                    $user->setFirstName($formData['firstName']);
+                    $user->setLastName($formData['lastName']);
+                    $user->setEmail($invite->getEmail());
+                    $user->setPlainPassword($formData['plainPassword']);
+
+                    if ($invite->getRole()) {
+                        $user->setRole($invite->getRole());
+                    } else {
+                        /** @var RoleModel $roleModel */
+                        $roleModel = $this->getModel('user.role');
+                        $role      = $roleModel->getRepository()->findOneBy([], ['id' => 'ASC']);
+                        if (null !== $role) {
+                            $user->setRole($role);
+                        }
+                    }
+
+                    $encoded = $model->checkNewPassword($user, $hasher, $user->getPlainPassword());
+                    $user->setPassword($encoded);
+
+                    $model->saveEntity($user);
+                    $model->markInviteUsed($invite);
+                    $this->addFlashMessage('mautic.user.invite.account_created', [], 'notice', 'flashes');
+
+                    return $this->redirectToRoute('login');
+                } catch (\Exception) {
```

Let's not catch all the exceptions. Please catch only the specific exception that you expect. We don't want to silently fail if the database is not accessible for example. We want to fail with a 500 error and log this incident to the logs. This way we wouldn't know about it.